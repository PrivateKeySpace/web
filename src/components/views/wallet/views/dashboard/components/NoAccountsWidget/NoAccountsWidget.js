import React from 'react'
import { compose, setDisplayName, pure } from 'recompose'
import { css } from 'aphrodite/no-important'
import tx from '../../../../../../../theme/styles'
import sx from './styles'
import { Button } from '../../../../../../elements'

const displayName = 'components/views/wallet/views/dashboard/components/NoAccountsWidget'
const enhance = compose(
  setDisplayName(displayName),
  pure
)

function NoAccountsWidget () {
  return (
    <div className={css(tx.mx2, tx.flex, tx.flexDirectionRow)}>
      <div className={css(tx.mr4, tx.pt1, sx.icon)}>
        <span className='lnr lnr-users' />
      </div>
      <div className={css(sx.info)}>
        <div className={css(tx.mb1, tx.textTransformUppercase, tx.fontWeight400, sx.noAccountsMessage)}>You don't have multisig accounts yet</div>
        <Button type='button' variant='success' disabled>Add Account</Button>
      </div>
    </div>
  )
}

export default enhance(NoAccountsWidget)
