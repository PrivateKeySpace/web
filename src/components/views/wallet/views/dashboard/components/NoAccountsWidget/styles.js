import { StyleSheet } from 'aphrodite/no-important'
import { colors } from '../../../../../../../theme/constants'

const styles = StyleSheet.create({
  icon: {
    fontSize: 30,
    color: colors.gray
  },
  noAccountsMessage: {
    fontSize: 12,
    color: colors.gray
  }
})

export default styles
