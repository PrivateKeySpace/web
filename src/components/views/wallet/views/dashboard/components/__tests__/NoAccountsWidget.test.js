import React from 'react'
import { StyleSheetTestUtils } from 'aphrodite'
import { render } from 'enzyme'
import NoAccountsWidget from '../NoAccountsWidget'

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection()
})

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
})

it('renders without crashing and matches snapshot', () => {
  const widget = render(
    <NoAccountsWidget />
  )
  expect(widget).toMatchSnapshot()
})
