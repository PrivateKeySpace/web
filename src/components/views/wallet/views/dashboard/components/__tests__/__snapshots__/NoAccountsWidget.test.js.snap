// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and matches snapshot 1`] = `
<div
  class="mx2_1o6grbo-o_O-flex_jro6t0-o_O-flexDirectionRow_qb00m9"
>
  <div
    class="mr4_kh8dwt-o_O-pt1_vsuzfi-o_O-icon_nkv9vo"
  >
    <span
      class="lnr lnr-users"
    >
      
    </span>
  </div>
  <div
    class=""
  >
    <div
      class="mb1_1tpeo8l-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-noAccountsMessage_1yz1w90"
    >
      You don't have multisig accounts yet
    </div>
    <button
      class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1pvg0dr-o_O-success_1dopc1j"
      type="button"
    >
      Add Account
    </button>
  </div>
</div>
`;
